pkg/core/blockchain.go

```diff
@@ -1657,12 +1657,12 @@ var (
 func (bc *Blockchain) initVerificationVM(ic *interop.Context, hash util.Uint160, witness *transaction.Witness) error {
 	v := ic.VM
 	if len(witness.VerificationScript) != 0 {
-		if witness.ScriptHash() != hash {
-			return ErrWitnessHashMismatch
-		}
```

Comment:
Is this change really needed?

Reply:
No, this was only to get a specific error in test. I think we can remove the second check now, while leaving the test.

Reply:
I don't see any test being changed, so we either need that change or don't need this one.